CurrencyConverterControl/Data/CurrencyDataProvider.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using CurrencyConverterControl.Model;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using System.Configuration;
+
+namespace CurrencyConverterControl.Data
+{
+    internal class CurrencyDataProvider : ICurrencyDataProvider
+    {
+        private static readonly HttpClient httpClient = new HttpClient();
+        string baseURl = "http://api.currencylayer.com/";
+        string access_key = "63da635cc4fff2e738aeb69cd8de0e91";
+        Dictionary<string, double> exchangeRates = new Dictionary<string, double>();
+        public CurrencyDataProvider()
+        {
+           
+        }
+
+        /// <summary>
+        /// This method is used to perfom the currency convertion
+        /// </summary>
+        /// <param name="currencyfrom">This parameter gets the value of the source currency</param>
+        /// <param name="currencyto">This parameter gets the value of the destination currency</param>
+        /// <param name="amount">This parameter gets the value which needs to be converted</param>
+        /// <returns></returns>
+        /// <exception cref="HttpRequestException"></exception>
+        public double Convert(string currencyfrom, string currencyto, double amount)
+        {
+            try
+            {
+                if (currencyfrom == currencyto) { return amount; }
+               
+               
+                    httpClient.DefaultRequestHeaders.Accept.Clear();
+                    httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
```

Comment:
You use the same settings for all requests. You can set it once in the static constructor.